projects/ng-components-docs/src/app/examples/components/table/table-examples.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnInit } from "@angular/core";
+import { TableHeaderListItem } from "lib/src/table/table.models";
+import { TableService } from "lib/src/table";
+
+interface TableObjectType {
+    foo: string;
+    bar: string;
+}
+
+@Component({
+    selector: "seb-table-examples",
+    templateUrl: "./table-examples.component.html",
+})
+export class TableExamplesComponent implements OnInit {
+    sortable: boolean;
+    selectable: boolean;
+
+    headerList: TableHeaderListItem<TableObjectType>[];
+    rows: TableObjectType[];
+
+    data: TableObjectType[] = [
+        { foo: "foo 1", bar: "bar 1" },
+        { foo: "foo 2", bar: "bar 2" },
+        { foo: "foo 3", bar: "bar 3" },
+    ];
+
+    constructor(private tableService: TableService<TableObjectType>) {}
+
+    ngOnInit() {
+        // Register your data with the table service.
+        // (Optional): Pass in your config. The service will return
+        // the headerList and rows to be displayed with the Table component.
+        this.tableService.registerDatasource(this.data);
+
+        // Subscribe to the current table and header list
+        this.tableService.currentTable.subscribe({
```

Comment:
i found a nice way to unsubscribe from obserables without creating an array checking if it has elements and unsubscribing from it.

```typescript
       import { takeUntil } from 'rxjs/operators';
       const notifier$: Subject<any> = new Subject();
        this.tableService.currentTable.pipe(takeUnitl(notifier$)).subscribe();
        
        ngOnDestroy(): void {
          this.notifier$.next();
        }
```

Reply:
Still looks a bit hacky but it's better than unsubscribing within array. I just wish angular would have a straightforward way of unsubscribing. Does this work with multiple subscriptions as well? I just use the same `takeUnitl(notifier$)` within every subscription and then `notifier$.next();` on destroy?

Reply:
there is many ways to do it, this one looks really clean and yes you can use the same to unsubscribe from all the subscriptions.
The best way to use observables is though async pipe but this case is only when you wanna subscribe and use the observable in the template, async pipe will handle the unsubscription by default

Reply:
ok added